notebooks/flightplanning_utils.py

```diff
@@ -0,0 +1,488 @@
+        )
+
+        if dist < dist_:
+            data.append(new_row)
+            dist_ = dist
+
+            pos = data[-1]
+        else:
+            dt = int(dt / 10)
+            assert dt > 0
+
+        loop += 1
+
+    return pd.DataFrame(data)
+
+
+def plot_trajectory(
+    lat1, lon1, lat2, lon2, trajectory: pd.DataFrame, ds: xr.Dataset
+) -> Figure:
+    """Plot the trajectory of an object
+
+    Args:
+        trajectory (pd.DataFrame): the trajectory of the object
+        ds (xr.Dataset): the dataset containing the wind field
+
+    Returns:
+        Figure: the figure
+    """
+
+    fig = Figure(figsize=(600, 600))
+    fig.canvas.header_visible = False
+    fig.canvas.footer_visible = False
+    fig.canvas.resizable = False
+    fig.set_dpi(1)
+
+    # lon1, lat1 = trajectory.iloc[0]["lon"], trajectory.iloc[0]["lat"]
+    # lon2, lat2 = trajectory.iloc[-1]["lon"], trajectory.iloc[-1]["lat"]
+
+    latmin, latmax = min(lat1, lat2), max(lat1, lat2)
+    lonmin, lonmax = min(lon1, lon2), max(lon1, lon2)
+
+    ax = plt.axes(
+        projection=ccrs.TransverseMercator(
+            central_longitude=(lonmax - lonmin) / 2,
+            central_latitude=(latmax - latmin) / 2,
+        )
+    )
+
+    wind_sample = 30
+
+    ax.set_extent([lonmin - 4, lonmax + 4, latmin - 2, latmax + 2])
+    ax.add_feature(OCEAN, facecolor="#d1e0e0", zorder=-1, lw=0)
+    ax.add_feature(LAND, facecolor="#f5f5f5", lw=0)
+    ax.add_feature(BORDERS, lw=0.5, color="gray")
+    ax.gridlines(draw_labels=True, color="gray", alpha=0.5, ls="--")
+    ax.coastlines(resolution="50m", lw=0.5, color="gray")
+
+    if ds is not None:
+        t = trajectory.ts.iloc[-1]
+        alt = trajectory.alt.iloc[-1]
+        """
+        self.wind_interpolator.plot_wind(
+            alt=alt,
+            lon_min=max(-180, lonmin - 4),
+            lon_max=min(+180, lonmax + 4),
+            lat_min=max(-90, latmin - 2),
+            lat_max=min(+90, latmax + 2),
+            t=int(t),
+            n_lat=180,
+            n_lon=720,
+            plot_wind=False,
+            plot_barbs=False,
+            ax=ax,
+        )
+        """
+
+    # great circle
+    ax.scatter(lon1, lat1, c="darkgreen", transform=ccrs.Geodetic())
+    ax.scatter(lon2, lat2, c="tab:red", transform=ccrs.Geodetic())
+
+    ax.plot(
+        [lon1, lon2],
+        [lat1, lat2],
+        label="Great Circle",
+        color="tab:red",
+        ls="--",
+        transform=ccrs.Geodetic(),
+    )
+
+    # trajectory
+    ax.plot(
+        trajectory.lon,
+        trajectory.lat,
+        color="tab:green",
+        transform=ccrs.Geodetic(),
+        linewidth=2,
+        marker=".",
+        label="Optimal",
+    )
+
+    ax.legend()
+
+    # Save it to a temporary buffer.
+    # buf = BytesIO()
+    # fig.savefig(buf, format="png")
+    # Embed the result in the html output.
+    # data = base64.b64encode(buf.getbuffer()).decode("ascii")
+
+    return fig
+
+
+if __name__ == "__main__":
```

Comment:
might be better to remove your custom local test.

Reply:
done